src/main/ml-modules/ext/query/structured-query-utils.xqy

```diff
@@ -0,0 +1,78 @@
+xquery version "1.0-ml";
+
+(:~
+ : Utility module from converting structured queries to/from JSON, XML, and cts:query
+ :)
+module namespace sq = "http://marklogic.com/mlpm/structured-query";
+
+import module namespace ast = "http://marklogic.com/appservices/search-ast"
+  at "/MarkLogic/appservices/search/ast.xqy";
```

Comment:
Same comment as on the service, these are all internal libs so what kind of version dependency are we adding here?

Reply:
Not sure.. That lib has been out there for a long while now, never had to update it so far..

Reply:
Unfortunately, I don't really see another way. And not sure if version dependency is any bigger than leaning on public interfaces..

Reply:
Got it. I'm ok using the lib but I just wanted to make sure we note any specific version dependency if this adds one.

Reply:
Looking around in the code a bit further I did see: https://github.com/marklogic-community/marklogic-geo-data-services/blob/b3e434014c451235ec773ec497e4eeb95f5d254b/src/main/ml-modules/ext/search/geo-search-util.xqy#L5

Wondering if I could lean on that instead.. and it is leaning on an internal lib as well..

Reply:
Maybe worth another quick call on Monday?

Reply:
It does look like there is some duplication between those two libraries. Perhaps we can just move the `from-json`, `to-json`, and `to-cts` functions to that lib?

Also maybe add a `json-to-cts` function so you can go directly from JSON structured query to a cts object in one call?

Reply:
I'll try to merge what I need into that geo utils lib, and will have a look at those tests..

Reply:
Tests added, libs merged. Is this resolved?

Reply:
Looks good to me.